Spigot-Server-Patches/0736-Better-Stats-API.patch

```diff
@@ -0,0 +1,211 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Thu, 20 May 2021 01:10:15 -0700
+Subject: [PATCH] Better Stats API
+
+
+diff --git a/src/main/java/io/papermc/paper/statistics/PaperStatistics.java b/src/main/java/io/papermc/paper/statistics/PaperStatistics.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..7683c4a3de811f1c252d192b625aed8b3e027145
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/statistics/PaperStatistics.java
+@@ -0,0 +1,87 @@
++package io.papermc.paper.statistics;
++
++import net.minecraft.core.IRegistry;
++import net.minecraft.resources.MinecraftKey;
++import net.minecraft.stats.Counter;
++import net.minecraft.stats.StatisticList;
++import net.minecraft.stats.StatisticWrapper;
++import net.minecraft.world.entity.EntityTypes;
++import net.minecraft.world.item.Item;
++import net.minecraft.world.level.block.Block;
++import org.bukkit.Keyed;
++import org.bukkit.Material;
++import org.bukkit.NamespacedKey;
++import org.bukkit.Registry;
++import org.bukkit.craftbukkit.util.CraftMagicNumbers;
++import org.bukkit.craftbukkit.util.CraftNamespacedKey;
++import org.bukkit.entity.EntityType;
++import org.jetbrains.annotations.NotNull;
++
++public final class PaperStatistics {
++
++    private PaperStatistics() {
++    }
++
++    @NotNull
++    public static Statistic<?> getPaperStatistic(net.minecraft.stats.Statistic<?> nmsStat) {
++        StatisticType<?> type = Registry.STATISTIC_TYPES.get(CraftNamespacedKey.fromMinecraft(IRegistry.STATS.getKey(nmsStat.getWrapper())));
++        Statistic<?> paperStat;
++        if (type == StatisticType.BLOCK_MINED) {
```

Comment:
Migrating to a switch case could cleanup things.
1 call for Material (ItemType/BlockType) with all cases that fall under that.
1 call for EntityType with all cases that fall under that.

Reply:
There aren't any enums there, so I'm not quite sure how to change this to a switch case.